core/client/fs/src/main/java/alluxio/client/block/stream/BlockWorkerDataWriter.java

```diff
@@ -102,6 +102,7 @@ public void writeChunk(final ByteBuf buf) throws IOException {
       }
     }
     long append = mBlockWriter.append(buf);
+    buf.release();
```

Comment:
Wrap the whole method internal content into a try{}finally{buf.release} block to make sure whatever happens inside the writeChunk operation, the buf can always be released.
Otherwise we may still memory leak when BlockWriter append buf failed or other exceptions thrown